### DISCUSSION TITLE
Clarify test262 coverage

### DISCUSSION THREAD

**Discussion by rwalle** [swc-project/swc#9231] [General]
swc_ecma_parser [documentation page](https://docs.rs/swc_ecma_parser/latest/swc_ecma_parser/) says:

> Passes almost all tests from [tc39/test262](https://github.com/tc39/test262)

Can we get the coverage of test262? I think the information is very important to potential users -- they want to know what "almost all" refers to and if this is ready for serious projects. It would be better to write out known limitations.

I actually tracked down the test in `crates/swc_ecma_parser/tests/test262.rs`. Looks like this file needs some maintenance -- if I comment out the first item `431ecef8c85d4d24.js` in `IGNORED_PASS_TESTS` the test still passes, but not `3dabeca76119d501.js` in the "annex B" area. The test directly skips anything that is in "ignored". I wonder if the test should be written in such a way that it expects these things to fail instead of ignoring them all together? This way, if a file can be parsed, it must be removed from the list

**Comment by kdy1** [comment]
All specs are implemented, excluding Annex B, which is an extension rather than a standard. The documentation should be updated. Can you send a PR?

**Reply by rwalle** [reply]
Sure, I'll try something in the next few days

**Reply by rwalle** [reply]
@kdy1 Looks like the documentation needs a bit more work -- my understanding is that `swc_ecma_parser` also support JSX. Could you confirm
* JSX is fully tested
* most TypeScript tests pass with exceptions [here](https://github.com/swc-project/swc/blob/83e75baaed098e5e4ee2015f6e76dceaf588b08c/crates/swc_ecma_parser/tests/typescript.rs#L123)

Also, could you address the question above -- it seems that many of the items in `IGNORED_PASS_TESTS` are outdated and come from 4 or 7 years ago ([git blame](https://github.com/swc-project/swc/blame/main/crates/swc_ecma_parser/tests/test262.rs#L26)). I would do this separately, but do you think this should use some clean up and stricter logic to avoid false negatives?
